### PR TITLE
Add latest Python versions to Travis config

### PR DESCRIPTION
**Pull request by ecederstrand**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,7 +27,8 @@ matrix:
   fast_finish: true
 
 install:
-  - pip install --upgrade pip setuptools wheel tox tox-travis
+  - python -m ensurepip
```
Comment (Midnighter) [inline .travis.yml]: The Travis Python build environment surely provides pip so this seems redundant.

File: .travis.yml [hunk 2]
```diff
@@ -44,9 +46,10 @@ stages:
 jobs:
   include:
     - stage: deploy
-      python: "3.6"
+      python: "3.8"
       install:
-        - pip install --upgrade pip setuptools wheel
+        - python -m ensurepip
```
Comment (Midnighter) [inline .travis.yml]: As above.

File: .travis.yml (hunk 1 above)
Comment (ecederstrand) [reply .travis.yml]: I added ensurepip in n attempt to fix https://travis-ci.org/opencobra/cobrapy/jobs/649362347?utm_medium=notification&utm_source=github_status. It apparently didn't fix the problem. I'll remove it again.

**Comment by cdiener** [comment]
I added CI fixes to https://github.com/opencobra/cobrapy/pull/929. After that one is merged you can just rebase on devel and everything should (hopefully) just work 🤞

**Comment by ecederstrand** [comment]
Rebased from develop just now. All checks are green.

File: tox.ini [hunk 3]
```diff
@@ -1,15 +1,15 @@
 [tox]
-envlist = pep8, isort, safety, py{27,35,36,37}
+envlist = pep8, isort, safety, py{35,36,37,38}
```
Comment (Midnighter) [inline tox.ini]:
```suggestion
envlist = pep8, isort, safety, py3{5,6,7,8,9}
```

File: tox.ini [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 
 [travis]
 os =
-    linux: pep8, isort, safety, py{27,35,36,37}
-    osx: py{27,35}
+    linux: pep8, isort, safety, py{35,36,37,38}
```
Comment (Midnighter) [inline tox.ini]: As above.

File: tox.ini [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
+    osx: py{35}
 python =
-    2.7: py27, pep8, safety
     3.5: py35, pep8, isort, safety
     3.6: py36
     3.7: py37
+    3.8: py38
```
Comment (Midnighter) [inline tox.ini]: Add line

```
3.9: py39
```

**Comment by ecederstrand** [comment]
I managed to delete my fork by accident. Oops. I think I'll just open a new PR.

Are you sure you want Python 3.9? It's not scheduled for release before October 2020: https://www.python.org/dev/peps/pep-0596/ I think Travis only has the `3.9-dev` target right now.

**Comment by Midnighter** [comment]
> Are you sure you want Python 3.9?

Well, that was the purpose of your PR, right? My other suggested changes only ensure that the correct tox environments are tested on Travis automatically.

**Comment by Midnighter** [comment]
Oh, if you don't have the branch any longer I will just merge this and add the relevant bits in another commit.

**Comment by ecederstrand** [comment]
My main purpose was to add 3.8. I just added 3.9-dev to see if it would work.

I've opened https://github.com/opencobra/cobrapy/pull/939 with the changes you suggested here. Feel free to do with either as you wish.

**Comment by Midnighter** [comment]
Cheers, closing this one then.

**Pull request closed without merging**